pkg/build/controller/pipeline_controller.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+// JenkinsPipelineServiceName is the name of the Jenkins service this pipeline
+// deploys.
+const JenkinsPipelineServiceName = "ci"
+
+// PipelineControllerFatalError represents the fatal error that might occur
+// during Jenkins deployment.
+type PipelineControllerFatalError struct {
+	// Reason the fatal error occurred
+	Reason string
+}
+
+// Error returns the error string for this fatal error
+func (e PipelineControllerFatalError) Error() string {
+	return fmt.Sprintf("fatal error processing BuildConfig: %s", e.Reason)
+}
+
+// PipelineController is the controller that handles deployment of new pipeline.
+type PipelineController struct {
+	PipelineInstantiator buildclient.PipelineInstantiator
+	KubeClient           kclient.Interface
+}
+
+// HandlePipeline handles new pipeline creation. This function will run when a
+// new build configuration is created with type JenkinsPipelineStrategy.
+// If the project where this configuration is create does not already have the
+// "jenkins" service, this will deploy a template which provision a new Jenkins
+// instance.
+func (c *PipelineController) HandlePipeline(bc *buildapi.BuildConfig) error {
```

Comment:
assuming you keep this, it's really HandleBuildConfig still, since that is what it is handling.